katz/src/main/kotlin/katz/data/Coproduct.kt

```diff
@@ -22,33 +22,25 @@ typealias CoproductKind<F, G, A> = HK<CoproductFG<F, G>, A>
 
 fun <F, G, A> CoproductKind<F, G, A>.ev(): Coproduct<F, G, A> = this as Coproduct<F, G, A>
 
-data class Coproduct<F, G, A>(val run: Either<HK<F, A>, HK<G, A>>) : CoproductKind<F, G, A> {
+data class Coproduct<F, G, A>(val CF: Comonad<F>, val CG: Comonad<G>, val run: Either<HK<F, A>, HK<G, A>>) : CoproductKind<F, G, A> {
 
     class F private constructor()
 
-    companion object {
```

Comment:
`Coproduct` could not have same companion object like `Kleisli` in order to provide their Comonad instances from implicits?